### PR TITLE
fix(dev): psycopg2-binary 2.9.1 does not work with Django 2.2

### LINKED ISSUE #28958: Downgrade psycopg2-binary on Python 3.8
psycopg 2.9.x is fundamentally incompatible with Django 2.2: https://github.com/psycopg/psycopg2/issues/1293

We'll have to stay on 2.8.x and find a way to get it working with Apple M1 machines.

### PR DESCRIPTION
**Pull request by armenzg**
For the incompatibility issue with Django visit [this issue][incompatible],

In this PR, we're going to downgrade to the compatible 2.8.6 version (thus, matching
the Python 3.6 version), removing the need to install `postgresql` (this means partially
reverting #28607) and instead installing a wheel I built on my machine that I have stored
in GC storage.

I built the wheel like this (see [root issue][issue] for details):
```shell
brew install postgresql
export CPPFLAGS="-I/opt/homebrew/opt/openssl@1.1/include"
export LDFLAGS="-L/opt/homebrew/opt/openssl@1.1/lib -L/opt/homebrew/opt/python@3.8/lib"
pip install psycopg2-binary==2.8.6
```

Fixes #28958

[incompatible]: https://github.com/psycopg/psycopg2/issues/1293
[issue]: https://github.com/psycopg/psycopg2/issues/1286

### CODE REVIEW

**Comment by armenzg** [comment]
For some odd reason, the URL sometimes doesn't work and requires authentication. I will wait until I can get IT to make the bucket public.

File: .github/workflows/development-environment.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
         # Xcode CLI & brew are already installed, thus, no need to call xcode-select install
         continue-on-error: true
         run: |
-          make prerequisites
+          brew bundle -q
```
Comment (armenzg) [inline .github/workflows/development-environment.yml:54]: I'm dropping the idea of having this target.

File: scripts/lib.sh [hunk 2]
```diff
@@ -129,6 +129,12 @@ install-py-dev() {
     # This helps when getsentry calls into this script
     cd "${HERE}/.." || exit
     echo "--> Installing Sentry (for development)"
+    if query-apple-m1; then
+        # This installs pyscopg-binary2 since there's no arm64 wheel
+        # This saves having to install postgresql on the Developer's machine + using flags
+        # https://github.com/psycopg/psycopg2/issues/1286
+        pip install "https://storage.googleapis.com/python-arm64-wheels/psycopg2_binary-2.8.6-cp38-cp38-macosx_11_0_arm64.whl"
```
Comment (joshuarli) [inline scripts/lib.sh:136]: Where/how is this being built and uploaded?

Comment (armenzg) [reply scripts/lib.sh:136]: *magic* JK :)

I had commented in the description of the PR:

```
brew install postgresql
export CPPFLAGS="-I/opt/homebrew/opt/openssl@1.1/include"
export LDFLAGS="-L/opt/homebrew/opt/openssl@1.1/lib -L/opt/homebrew/opt/python@3.8/lib"
pip install psycopg2-binary==2.8.6
```

I then grab the wheel from the cache.

**Review by joshuarli — APPROVED** [review]

**Comment by joshuarli** [comment]
Oh. @armenzg you had the target branch set to something else. Guess I gotta merge that too then.

**Pull request merged**